partition: add doc for global index (#17979)
pingcap/docs: system-variables.md

### tidb_enable_global_index <span class="version-mark">New in v7.6.0</span>

> **Warning:**
>
> The feature controlled by this variable is an experimental feature. It is not recommended that you use it in the production environment. This feature might be changed or removed without prior notice. If you find a bug, you can report an [issue](https://github.com/pingcap/tidb/issues) on GitHub.

- Scope: SESSION | GLOBAL
- Persists to cluster: Yes
- Applies to hint [SET_VAR](/optimizer-hints.md#set_varvar_namevar_value): No
- Type: Boolean
- Default value: `OFF`
- Possible values: `OFF`, `ON`

> - This variable controls whether to support creating `Global indexes` for partitioned tables. When this variable is enabled, TiDB could create unique indexes that do not contain all columns in the partition expressions.

Reviewer: ```suggestion
- This variable controls whether to support creating [`Global indexes`](/partitioned-table.md#global-index) for partitioned tables. When this variable is enabled, TiDB could create unique indexes that do not contain all columns in the partition expressions.
```

> - This variable controls whether to support creating [`global indexes`](/partitioned-table.md#global-indexes) for partitioned tables. When this variable is enabled, TiDB creates unique indexes that do not contain all columns in the partition expressions.

Reviewer 2: ```suggestion
- This variable controls whether to support creating [global indexes](/partitioned-table.md#global-indexes) for partitioned tables. When this variable is enabled, TiDB creates unique indexes that do not contain all columns in the partition expressions.
```

> - This variable controls whether to support creating [`Global indexes`](/partitioned-table.md#global-index) for partitioned tables. When this variable is enabled, TiDB could create unique indexes that do not contain all columns in the partition expressions.

Reviewer 2: ```suggestion
- This variable controls whether to support creating [`global indexes`](/partitioned-table.md#global-indexes) for partitioned tables. When this variable is enabled, TiDB creates unique indexes that do not contain all columns in the partition expressions.
```
